### PR TITLE
Check for valid DirectionsRoute in RouteRefresh

### PR DESCRIPTION
**Pull request by danesfeder**
Found while testing:
```
E/AndroidRuntime: FATAL EXCEPTION: mapbox_navigation_thread
    Process: com.mapbox.mapboxsdk.downstream.testapp, PID: 1955
    java.lang.NumberFormatException: Invalid int: "null"
        at java.lang.Integer.invalidInt(Integer.java:138)
        at java.lang.Integer.parseInt(Integer.java:358)
        at java.lang.Integer.parseInt(Integer.java:334)
        at java.lang.Integer.valueOf(Integer.java:525)
        at com.mapbox.services.android.navigation.v5.navigation.RouteRefresh.refresh(RouteRefresh.java:80)
        at com.mapbox.services.android.navigation.v5.navigation.RouteRefresh.refresh(RouteRefresh.java:74)
        at com.mapbox.services.android.navigation.v5.navigation.RouteRefresher.refresh(RouteRefresher.java:33)
        at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorRunnable.process(RouteProcessorRunnable.java:66)
        at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorRunnable.run(RouteProcessorRunnable.java:46)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:148)
        at android.os.HandlerThread.run(HandlerThread.java:61)
```

## Description

We should protect against this error.  It seems it was produced while sitting in a stale navigation state for a while.

- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Implementation

Validity check that will fire error callback with error message when invalid.

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed)
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteRefresh.java [hunk 1]
```diff
@@ -75,11 +77,23 @@ public void refresh(RouteProgress routeProgress, RefreshCallback refreshCallback
...
       .routeIndex(Integer.valueOf(directionsRoute.routeIndex()))
       .legIndex(legIndex)
       .accessToken(accessToken)
       .build().enqueueCall(new RouteRefreshCallback(directionsRoute, legIndex, refreshCallback));
   }
+
+  private boolean isInvalid(DirectionsRoute directionsRoute, RefreshCallback refreshCallback) {
+    String requestUuid = directionsRoute.routeOptions().requestUuid();
+    if (TextUtils.isEmpty(requestUuid) || directionsRoute.routeIndex() == null) {
+      refreshCallback.onError(new RefreshError("RouteProgress passed has invalid DirectionsRoute"));
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteRefresh.java]: NIT Could we extract this magic number into a constant?

Comment (Guardiola31337) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteRefresh.java]: Not sure if we should report this as an "error" until we know why the Refresh API is returning `null` after some time stale. Should we maybe stop the periodic refresh for this scenario? 🤔

Comment (danesfeder) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteRefresh.java]: I think it's fine to send the error, our behavior is to fail silently:
```
  @Override
  public void onError(RefreshError error) {
    Timber.w(error.getMessage());
    routeRefresher.updateIsChecking(false);
  }
```
I extracted the magic number 👍

**Comment by danesfeder** [comment]
> Wondering why is that happening though

Yeah, good to protect against it for now.  But we should keep an eye on this.

**Pull request merged**